cumulus_library/template_sql/statistics/count.sql.jinja

```diff
@@ -35,65 +38,60 @@ CREATE TABLE {{ table_name }} AS (
         {%- endif %}
     ),
     {% endif %}
-    powerset AS (
+    null_replacement AS (
         SELECT
-            count(DISTINCT subject_ref) AS cnt_subject,
-            {%- if fhir_resource=='condition' %}
-            count(DISTINCT condition_ref) AS cnt_condition,
-            {%- elif fhir_resource=='document' %}
-            count(DISTINCT document_ref) AS cnt_document,
-            {%- elif fhir_resource=='encounter' %}
-            count(DISTINCT encounter_ref) AS cnt_encounter,
-            {%- elif fhir_resource=='observation' %}
-            count(DISTINCT observation_ref) AS cnt_observation,
+            subject_ref,
+            {%- if secondary_resource(fhir_resource) !='None' %}
+            {{ secondary_resource(fhir_resource) }}_ref,
             {%- endif -%}
             {% for col in table_cols %}
-            "{{ col }}"
-            {%- if not loop.last -%}
-            , 
-            {%- endif -%}
+            coalesce(cast("{{ col }}" AS varchar), 'missing-or-null') AS "{{ col }}"
+            {{- comma_delineate(loop) }}
             {%- endfor %}
         {%- if filter_resource %}
         FROM filtered_table
         {%- else  %}
         FROM {{ source_table }}
         {% endif %}
+    ),
+
+    powerset AS (
```

Comment:
It's actually the inverse in many cases - small actions with fewer joins are easier for athena to parallelize. Complex joins cause a lot of upfront overhead since it gets harder to farm things out to worker nodes.